src/main.py

```
@@ -13,11 +17,53 @@
+                logging.info("Bot has access to: %s", channel.name)
+
+
+@bot.event
+async def on_error(error, *args, **kwargs):
+    if isinstance(args[0], discord.HTTPException) and args[0].status == 429:
+        # Rate limit encountered
+        retry_after = args[0].headers.get("Retry-After")
+        if not retry_after:
+            retry_after = 5
+            
+        logging.warning("Rate limit encountered. Retrying after %s seconds.", retry_after)
```

Ruff (0.8.2)

`logging` may be undefined, or defined from star imports

(F405)